corehq/apps/locations/permissions.py

```diff
@@ -319,6 +319,19 @@ def user_can_access_any_location_id(domain, user, location_ids):
             .exists())
 
 
+def user_can_access_other_user(domain, user, other_user):
+    if user.has_permission(domain, 'access_all_locations'):
+        return True
+
+    accessible_location_ids = (SQLLocation.objects .accessible_to_user(domain, user) .values_list('id', flat=True))
+
+    other_user_locations = other_user.sql_locations
```

Comment:
this function is kind of funky in that when you do have locations, it returns a queryset. when you don't have locations it returns an empty array which seems like it could cause errors pretty easily if you do something like `user.sql_locations.values_list`